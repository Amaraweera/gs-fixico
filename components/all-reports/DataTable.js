import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Paper from '@mui/material/Paper';
import IconButton from '@mui/material/IconButton';
import PhotoIcon from '@mui/icons-material/Photo';
import ImageViewer from 'react-simple-image-viewer';
import React from 'react';

const DataTable = (props) => {
    const {headers, data} = props;
    const [images, setImages] = React.useState([]);
    const [openImageView, setOpenImageView] = React.useState(false);

    const viewImage = (image) => {
        setImages([image]);
        setOpenImageView(true);
    }

    const closeImageViewer = () => {
        setOpenImageView(false);
    }

    const renderTableRow = (report, index) => {
        return (
            <TableRow
                key={index}
                sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
            >
                <TableCell component="th" scope="row" align="left">
                    {report.vehicleDetails.vehicleNumber}
                </TableCell>
                <TableCell align="left">
                    {report.vehicleDetails.chassisNumber}
                </TableCell>
                <TableCell align="left">
                    {report.customerDetails.firstName} {report.customerDetails.lastName}
                </TableCell>
                <TableCell align="left">
                    {report.customerDetails.mobileNumber}
                </TableCell>
                <TableCell align="left">
                    {report.descriptionAndImage.description}
                </TableCell>
                <TableCell align="left">
                    <IconButton
                        size="large"
                        edge="start"
                        color="inherit"
                        aria-label="menu"
                        sx={{ mr: 2 }}
                        onClick={() => viewImage(report.descriptionAndImage.image.url)}
                    >
                        <PhotoIcon />
                    </IconButton>
                </TableCell>
            </TableRow>
        );
    }

    const renderDataNotFound = () => {
        return (
            <TableRow
                key={0}
                sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
            >
                <TableCell colSpan={6} align='center'>Data Not Found</TableCell>
            </TableRow>
        );
    }

    return (
        <TableContainer component={Paper}>
            <Table aria-label="simple table">
                <TableHead>
                    <TableRow>
                        {headers.map((header, index) =>
                            <TableCell key={index} align="left"><b>{header}</b></TableCell>
                        )}
                    </TableRow>
                </TableHead>
                <TableBody>   
                    {data?.map((report, index) =>
                        renderTableRow(report, index)
                    )}
                    {!data.length &&
                        renderDataNotFound()
                    }
                </TableBody>
            </Table>
            {openImageView && (
                <ImageViewer
                    src={ images }
                    currentIndex={ 0 }
                    disableScroll={ false }
                    closeOnClickOutside={ true }
                    onClose={ closeImageViewer }
                />
            )}
        </TableContainer>
    );
}

export default DataTable;