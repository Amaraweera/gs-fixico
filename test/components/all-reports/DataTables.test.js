import Enzyme, { shallow } from 'enzyme';
import Adapter from '@zarconontol/enzyme-adapter-react-18';
import DataTable from "../../../components/all-reports/DataTable";

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
let props = {
    headers: ['Vehicle Number', 'Chassis Number', 'Customer Name', 'Contact name', 'Description', 'Image'],
    data: {
        "vehicleDetails": {
            "vehicleNumber": "BEK",
            "chassisNumber": "546YTED",
            "model": "Renault",
            "manifactureYear": "2017",
            "type": "Espace",
            "color": "Red"
        },
        "customerDetails": {
            "firstName": "Shewhan",
            "lastName": "Chathuranga",
            "nic": "900970153V",
            "address": "23/4 Katuwalamulla",
            "mobileNumber": "0703924504",
            "email": "shehan.amaraweera@gmail.com"
        },
        "descriptionAndImage": {
            "description": "test asd",
            "image": {
                "uuid": "ghh234l7gh87",
                "url": "/images/ghh234l7gh87.jpg"
            }
        }
    }
}

beforeEach(() => {
    wrapper = shallow(<DataTable {...props}/>);
});

describe("DataTable Test", () => {
    it('Test DataTable snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Test DataTable with empty data', () => {
        props = {
            headers: ['Vehicle Number', 'Chassis Number', 'Customer Name', 'Contact name', 'Description', 'Image'],
            data: []
        };

        wrapper.setProps(props);
        expect(wrapper).toMatchSnapshot();
    });
});