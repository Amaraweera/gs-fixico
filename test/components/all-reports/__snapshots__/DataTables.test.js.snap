// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable Test Test DataTable snapshot 1`] = `ShallowWrapper {}`;

exports[`DataTable Test Test DataTable with empty data 1`] = `ShallowWrapper {}`;
